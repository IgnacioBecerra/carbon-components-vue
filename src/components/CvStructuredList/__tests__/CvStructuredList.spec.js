import { shallowMount, mount } from '@vue/test-utils';
import { CvStructuredList, CvStructuredListItem } from '../';

const DefaultId = 'test-1';

describe('CvStructuredList', () => {
  // ***************
  // SNAPSHOT TESTS
  // ***************
  it('should render correctly when not selectable and not condensed', async () => {
    const propsData = { DefaultId, selectable: false, condensed: false };
    const wrapper = shallowMount(CvStructuredList, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly when selectable', async () => {
    const propsData = { DefaultId, selectable: true, condensed: false };
    const wrapper = shallowMount(CvStructuredList, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly when condensed', async () => {
    const propsData = { DefaultId, selectable: false, condensed: true };
    const wrapper = shallowMount(CvStructuredList, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly when selectable and condensed', async () => {
    const propsData = { DefaultId, selectable: true, condensed: true };
    const wrapper = shallowMount(CvStructuredList, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render `headings` slot correctly', async () => {
    const propsData = { DefaultId, selectable: true, condensed: true };
    const wrapper = shallowMount(CvStructuredList, {
      propsData,
      slots: {
        headings: '<div class="headings-slot-test">Headings slot test</div>',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render `items` slot correctly', async () => {
    const propsData = { DefaultId, selectable: true, condensed: true };
    const wrapper = shallowMount(CvStructuredList, {
      propsData,
      slots: {
        items:
          '<ul class="items-slot-test"><li>Test 1.1</li><li>Test 1.2</li></ul>',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  // ***************
  // FUNCTIONAL TESTS
  // ***************
  it('pass on change events', async () => {
    const propsData = { selectable: true, condensed: false };
    const wrapper = mount(CvStructuredList, {
      propsData,
      slots: {
        items: CvStructuredListItem,
      },
    });
    const inputNode = wrapper.find('input');
    await inputNode.trigger('change');

    expect(wrapper.emitted('change')).toBeTruthy();
  });
});
