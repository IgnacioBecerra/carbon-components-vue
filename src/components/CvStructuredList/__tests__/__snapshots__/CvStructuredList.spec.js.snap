// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvStructuredList should render \`headings\` slot correctly 1`] = `
<div class="bx--structured-list bx--structured-list--selection bx--structured-list--condensed" data-structured-list="true" defaultid="test-1">
  <div class="bx--structured-list-thead">
    <div class="bx--structured-list-row bx--structured-list-row--header-row">
      <div class="headings-slot-test">Headings slot test</div>
      <div class="bx--structured-list-th"></div>
    </div>
  </div>
  <div class="bx--structured-list-tbody"></div>
</div>
`;

exports[`CvStructuredList should render \`items\` slot correctly 1`] = `
<div class="bx--structured-list bx--structured-list--selection bx--structured-list--condensed" data-structured-list="true" defaultid="test-1">
  <div class="bx--structured-list-thead">
    <div class="bx--structured-list-row bx--structured-list-row--header-row">
      <div class="bx--structured-list-th"></div>
    </div>
  </div>
  <div class="bx--structured-list-tbody">
    <ul class="items-slot-test">
      <li>Test 1.1</li>
      <li>Test 1.2</li>
    </ul>
  </div>
</div>
`;

exports[`CvStructuredList should render correctly when condensed 1`] = `
<div class="bx--structured-list bx--structured-list--condensed" data-structured-list="false" defaultid="test-1">
  <div class="bx--structured-list-thead">
    <div class="bx--structured-list-row bx--structured-list-row--header-row">
      <!--v-if-->
    </div>
  </div>
  <div class="bx--structured-list-tbody"></div>
</div>
`;

exports[`CvStructuredList should render correctly when not selectable and not condensed 1`] = `
<div class="bx--structured-list" data-structured-list="false" defaultid="test-1">
  <div class="bx--structured-list-thead">
    <div class="bx--structured-list-row bx--structured-list-row--header-row">
      <!--v-if-->
    </div>
  </div>
  <div class="bx--structured-list-tbody"></div>
</div>
`;

exports[`CvStructuredList should render correctly when selectable 1`] = `
<div class="bx--structured-list bx--structured-list--selection" data-structured-list="true" defaultid="test-1">
  <div class="bx--structured-list-thead">
    <div class="bx--structured-list-row bx--structured-list-row--header-row">
      <div class="bx--structured-list-th"></div>
    </div>
  </div>
  <div class="bx--structured-list-tbody"></div>
</div>
`;

exports[`CvStructuredList should render correctly when selectable and condensed 1`] = `
<div class="bx--structured-list bx--structured-list--selection bx--structured-list--condensed" data-structured-list="true" defaultid="test-1">
  <div class="bx--structured-list-thead">
    <div class="bx--structured-list-row bx--structured-list-row--header-row">
      <div class="bx--structured-list-th"></div>
    </div>
  </div>
  <div class="bx--structured-list-tbody"></div>
</div>
`;
